CONTRIBUTING.md

```diff
@@ -20,6 +20,16 @@ again.
 The Python part of the project will follow [Google Python style guide](http://google.github.io/styleguide/pyguide.html). We provide a [yapf](https://github.com/google/yapf) configuration file to help contributors auto-format their code to adopt the Google Python style. Also, it is encouraged to lint python docstrings by [docformatter](https://github.com/myint/docformatter).
 
 The frontend part of the project uses [prettier](https://prettier.io/) for formatting, read [frontend/README.md#code-style](frontend/README.md#code-style) for more details.
+## Unit Testing Best Practices
+
+* Testing via Public APIs
+
+### Golang
+* Put your tests in a different package: Moving your test code out of the package allows you to write tests as though you were a real user of the package. You cannot fiddle around with the internals,
```

Comment:
What's the usual convention of doing this? Can you point to an example?

Do we keep them in the same folder but add a package suffix of `.test`? or do we put it in a sub-folder?

Reply:
Usually it is the same folder and add a pacakge suffix of _test

Reply:
Added example